cmds/core/dd/dd.go

```diff
@@ -70,182 +68,44 @@ var flagMap = map[string]bitClearAndSet{
 
 var allowedFlags = os.O_TRUNC | os.O_SYNC
 
-// intermediateBuffer is a buffer that one can write to and read from.
-type intermediateBuffer interface {
-	io.ReaderFrom
-	io.WriterTo
-}
-
-// chunkedBuffer is an intermediateBuffer with a specific size.
-type chunkedBuffer struct {
-	outChunk int64
-	length   int64
-	data     []byte
-	flags    int
-}
-
-// newChunkedBuffer returns an intermediateBuffer that stores inChunkSize-sized
-// chunks of data and writes them to writers in outChunkSize-sized chunks.
-func newChunkedBuffer(inChunkSize int64, outChunkSize int64, flags int) intermediateBuffer {
-	return &chunkedBuffer{
-		outChunk: outChunkSize,
-		length:   0,
-		data:     make([]byte, inChunkSize),
-		flags:    flags,
-	}
-}
-
-// ReadFrom reads an inChunkSize-sized chunk from r into the buffer.
-func (cb *chunkedBuffer) ReadFrom(r io.Reader) (int64, error) {
-	n, err := r.Read(cb.data)
-	cb.length = int64(n)
-
-	// Convert to EOF explicitly.
-	if n == 0 && err == nil {
-		return 0, io.EOF
+func dd(r io.Reader, w io.Writer, inBufSize, outBufSize int64, bytesWritten *int64, flags int) error {
```

Comment:
Looks like flags variable is not used. 
And do we need bytesWritten as a pointer, could we have (int, error) as a return type?

Reply:
I removed flags from the dd func.
You need that pointer because of the way the progress package is designed. The bytesWritten variable is monitored by a goroutine in the progress package. 

I don't much like that design, but fixing it is out of scope of this change ...